### PR TITLE
tests: adds support for testing multiple image types

### PR DESCRIPTION
**Pull request by andrewschoen**
Now the CI tests can take input for each testing scenario specifying which image to build and test. For example, now the ``ceph_ansible2.1-jewel-centos7_cluster`` testing scenario is actually building and deploying a centos7 jewel image.

### CODE REVIEW

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-jewel-xenial_cluster

File: tests/tox.sh [hunk 1]
```diff
@@ -33,12 +33,20 @@ if [ "${containers_to_remove}" ]; then
     docker rm -f $@ ${containers_to_remove} || echo failed to remove containers
 fi
 
+# copy the files to the root for the
+# types of images we're going to build
+mkdir -p {base,daemon,demo}
+cp -Lrv ceph-releases/$CEPH_RELEASE/$IMAGE_DISTRO/base/* base
```
Comment (leseb) [inline tests/tox.sh]: Can we add `|| true` to this one? With Kraken this dir won't exist :)

Comment (andrewschoen) [inline tests/tox.sh]: Sure thing, fixed.

**Comment by andrewschoen** [comment]
jenkins test this please

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-jewel-xenial_cluster

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-jewel-xenial_cluster

File: tests/tox.sh [hunk 2]
```diff
@@ -33,6 +33,14 @@ if [ "${containers_to_remove}" ]; then
     docker rm -f $@ ${containers_to_remove} || echo failed to remove containers
 fi
 
+# copy the files to the root for the
+# types of images we're going to build
+mkdir -p {base,daemon,demo}
+# starting with kraken, the base image does not exist
+cp -Lrv ceph-releases/$CEPH_RELEASE/$IMAGE_DISTRO/base/* base || true
+cp -Lrv ceph-releases/$CEPH_RELEASE/$IMAGE_DISTRO/daemon/* daemon
+cp -Lrv ceph-releases/$CEPH_RELEASE/$IMAGE_DISTRO/demo/* demo
```
Comment (leseb) [inline tests/tox.sh]: sorry I forgot about this, chances are 'demo' might get included into daemon as well for kraken. Can you add `|| true` as well? Once you're done, feel free to merge. :)

**Pull request merged**